packages/framework/src/Model/Order/Messenger/HeurekaPlacedOrderMessageHandler.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrameworkBundle\Model\Order\Messenger;
+
+use Exception;
+use Psr\Log\LoggerInterface;
+use Shopsys\FrameworkBundle\Model\Order\OrderFacade;
+use Symfony\Component\Messenger\Handler\MessageHandlerInterface;
+
+class HeurekaPlacedOrderMessageHandler implements MessageHandlerInterface
+{
+    /**
+     * @param \Psr\Log\LoggerInterface $logger
+     * @param \Shopsys\FrameworkBundle\Model\Order\OrderFacade $orderFacade
+     */
+    public function __construct(
+        protected readonly LoggerInterface $logger,
+        protected readonly OrderFacade $orderFacade,
+    ) {
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Order\Messenger\PlacedOrderMessage $placedOrderMessage
+     */
+    public function __invoke(PlacedOrderMessage $placedOrderMessage): void
+    {
+        try {
+            $orderSent = $this->orderFacade->sendHeurekaOrderInfo($placedOrderMessage->orderId);
+
+            if ($orderSent === true) {
+                $this->logger->info('Order successfully sent to Heureka', [
+                    'orderId' => $placedOrderMessage->orderId,
+                ]);
+            }
+        } catch (Exception $exception) {
+            $this->logger->error('Sending order to Heureka failed', [
+                'orderId' => $placedOrderMessage->orderId,
+                'exception' => $exception,
+            ]);
```

Comment:
There could be some network error or something, should we not try a few times to send it again after some time? Like half an hour or by cron?

Reply:
With the messenger, we have now great powers :blush: 254428bbd

To explain that, there is an inbuilt mechanism - when an exception is thrown during the message processing, the message is re-enqueued three times by default with 1, 2, and 4-second delay (I have configured longer delays in my solution, see the [docs](https://symfony.com/doc/current/messenger.html#retries-failures)).

What is really great and works out of the box is when there are multiple handlers for one message and one of them fails, the message is still re-enqueued but it is then processed just by the handlers that failed (e.g. when sending to heureka fails but email is sent successfully, it will not try to send the email again). This is thanks to the [stamps system](https://symfony.com/doc/current/components/messenger.html#adding-metadata-to-messages-envelopes), `HandledStamp` here in particular.

Don't you just love it? :heart_eyes: